Src/DependencyCollector/Shared/Implementation/HttpDiagnosticSourceListener.cs

```diff
@@ -47,15 +45,22 @@ public void OnNext(KeyValuePair<string, object> value)
         {
             try
             {
-                dynamic propertyBag = value.Value;                        
```

Comment:
Since you are removing use of dynamic, you can try removing the Microsoft.CSharp reference in the csproj. I added that to use dynamic